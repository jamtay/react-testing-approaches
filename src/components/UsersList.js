import { formatUrl } from "../utils/urlFormatter";

/**
 * Component to display a list of user's details in a list
 * @param data: Array of users including id, name, email, and website
 */
const UsersList = ({ data }) => {
  return (
    <>
      <h1>Users list</h1>
      <ul>
        {data.map(user => (
          <li key={user.id}>
            Name: {user.name}, Email: {user.email}, Website:{" "}
            {formatUrl(user.website)}
          </li>
        ))}
      </ul>
    </>
  );
};

export default UsersList;
