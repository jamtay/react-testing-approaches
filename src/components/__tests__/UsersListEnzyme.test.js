// Test the UsersList component using enzyme
import React from "react";
import { shallow } from "enzyme";
import UsersList from "../UsersList";

const FAKE_USERS = [
  {
    id: 1,
    name: "First Name",
    email: "myemail1@email.com",
    website: "website.com"
  },
  {
    id: 2,
    name: "Second Name",
    email: "myemai21@email.com",
    website: "www.website2.com"
  },
  {
    id: 3,
    name: "Third Name",
    email: "myemail3@email.com",
    website: "website3.com"
  }
];

describe("<UsersList/> component test with enzyme", () => {
  it("should match stored snapshot", () => {
    const wrapper = shallow(<UsersList data={FAKE_USERS} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the users list heading correctly", () => {
    const wrapper = shallow(<UsersList data={FAKE_USERS} />);
    expect(wrapper.find("h1").text()).toEqual("Users list");
  });

  it("should render the users list from the supplied data", () => {
    const wrapper = shallow(<UsersList data={FAKE_USERS} />);

    const listItems = wrapper.find("li");
    expect(listItems).toHaveLength(3);
    expect(listItems.at(0).text()).toEqual(
      "Name: First Name, Email: myemail1@email.com, Website: www.website.com"
    );
    expect(listItems.at(1).text()).toEqual(
      "Name: Second Name, Email: myemai21@email.com, Website: www.website2.com"
    );
    expect(listItems.at(2).text()).toEqual(
      "Name: Third Name, Email: myemail3@email.com, Website: www.website3.com"
    );
  });
});
