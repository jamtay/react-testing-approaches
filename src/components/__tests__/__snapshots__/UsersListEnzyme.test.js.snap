// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersList/> component test with enzyme should match stored snapshot 1`] = `
<Fragment>
  <h1>
    Users list
  </h1>
  <ul>
    <li
      key="1"
    >
      Name: 
      First Name
      , Email: 
      myemail1@email.com
      , Website:
       
      www.website.com
    </li>
    <li
      key="2"
    >
      Name: 
      Second Name
      , Email: 
      myemai21@email.com
      , Website:
       
      www.website2.com
    </li>
    <li
      key="3"
    >
      Name: 
      Third Name
      , Email: 
      myemail3@email.com
      , Website:
       
      www.website3.com
    </li>
  </ul>
</Fragment>
`;
